Solutions/Web Session Essentials/Hunting Queries/EmptyUserAgent.yaml

```diff
@@ -0,0 +1,48 @@
+id: 69e53015-a309-4a8f-a94d-df61a9217e2f
+name: Empty User Agent Detected (ASIM Web Session)
+description: |
+  'Rule helps to detect empty user agent indicating web browsing activity by an unusual process other than a web browser. It's important to analyze the context and patterns associated with web logs containing empty user agents.'
+tags:
+  - Schema: WebSession
+    SchemaVersion: 0.2.6
+requiredDataConnectors: []
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1190
+  - T1133
+query: |
+  let lookBack = 1d;
+  _Im_WebSession(starttime=ago(lookBack), eventresult='Success')
+  | where isempty(HttpUserAgent)
+  | summarize Count=count(), EventStartTime= min(TimeGenerated), EventEndTime=max(TimeGenerated) by SrcIpAddr, SrcUsername, SrcHostname, Url, DstIpAddr, DstPortNumber
+  | extend Name = iif(SrcUsername contains "@", tostring(split(SrcUsername,'@',0)[0]),SrcUsername), UPNSuffix = iif(SrcUsername contains "@",tostring(split(SrcUsername,'@',1)[0]),"")
+  | extend IP_0_Address = SrcIpAddr
+  | extend IP_1_Address = DstIpAddr
+  | extend Account_0_Name = Name
+  | extend Account_0_UPNSuffix = UPNSuffix
+  | extend URL_0_Url = Url
+  | extend Host_0_HostName = SrcHostname
+  | order by Count desc
+entityMappings:
+  - entityType: URL
+    fieldMappings:
+      - identifier: Url
+        columnName: Url
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: SrcIpAddr
+      - identifier: Address
```

Comment:
This should be a separate entity. Same Identifier cannot multiple values under same entity. However, you can have multiple entities of the same kind.

Reply:
this is done